tools/ClassGen/NodeGen.cpp

```diff
@@ -381,32 +389,53 @@ int main(int argc, char **argv) {
       .dataParallel()
       .setDocstring("Clip range of inputs to lie in [Min, Max].");
 
-  BB.newNode("CmpLTE")
+  BB.newNode("CmpEQ")
       .addInput("LHS")
       .addInput("RHS")
       .addResultFromCtorArg()
       .dataParallel()
-      .setDocstring("Performs CmpLTE on the LHS and RHS operands. Generates a "
-                    "mask that's consumed by the select instruction. The "
-                    "format of the result is target- and type-specific.");
+      .setDocstring("Performs an element-wise EQUAL comparison between the "
+                    "LHS and RHS operands.");
 
-  BB.newNode("CmpEQ")
+  BB.newNode("CmpNEQ")
       .addInput("LHS")
       .addInput("RHS")
       .addResultFromCtorArg()
       .dataParallel()
-      .setDocstring("Performs an element-wise equal comparison on the LHS and "
-                    "RHS operands. Inputs must be integer.");
+      .setDocstring("Performs an element-wise NOT EQUAL comparison between "
+                    "the LHS and RHS operands.");
 
   BB.newNode("CmpLT")
       .addInput("LHS")
       .addInput("RHS")
       .addResultFromCtorArg()
       .dataParallel()
-      .setDocstring(
-          "Compares X and Y element wise sets Dest[i] true if LHS[i] < "
-          "RHS[i] otherwise false. Final result is a mask consumed by "
-          "Select, ONNX Where, operator.");
+      .setDocstring("Performs an element-wise LESS THAN comparison between "
+                    "the LHS and RHS operands.");
+
+  BB.newNode("CmpLTE")
+      .addInput("LHS")
+      .addInput("RHS")
+      .addResultFromCtorArg()
+      .dataParallel()
+      .setDocstring("Performs an element-wise LESS THAN OR EQUAL comparison "
+                    "between the LHS and RHS operands.");
+
+  BB.newNode("CmpGT")
+      .addInput("LHS")
+      .addInput("RHS")
+      .addResultFromCtorArg()
+      .dataParallel()
+      .setDocstring("Performs an element-wise GREATER THAN comparison between "
+                    "the LHS and RHS operands.");
+
+  BB.newNode("CmpGTE")
+      .addInput("LHS")
+      .addInput("RHS")
+      .addResultFromCtorArg()
+      .dataParallel()
+      .setDocstring("Performs an element-wise GREATER THAN OR EQUAL comparison "
+                    "between the LHS and RHS operands.");
```

Comment:
Why do we need these? We can always just swap the LHS and RHS and use CmpLT and CmpLTE instead.

Reply:
You're right this was kind of dumb. I removed the node/instruction definitions/implementations for `CmpGT` and `CmpGTE` but I left instead in Graph.h/Graph.cpp two convenience functions:
```
  CmpLTNode *createCmpGT(llvm::StringRef name, NodeValue LHS, NodeValue RHS);
  CmpLTENode *createCmpGTE(llvm::StringRef name, NodeValue LHS, NodeValue RHS);
```
to create `CmpLT` and `CmpLTE` nodes with swapped inputs.